libsignal-service/src/websocket.rs

```diff
@@ -151,6 +151,17 @@ impl<WS: WebSocketService> SignalWebSocketProcess<WS> {
         }
     }
 
+    fn next_request_id(&self) -> u64 {
+        use rand::Rng;
+        let mut rng = rand::thread_rng();
+        loop {
+            let id = rng.gen();
```

Comment:
A matter of taste, but for expressivity, you can express that as

"please continue generating an id up until you find one that is not part of outgoing request"

so

```
iter::repeat_with(|| rng.gen()).filter(|id| !self.outgoing_request_map.contains_key(&id)).next()
```

Problem is that it returns an optional, but we might be able to unwrap here

Reply:
I was also thinking on `while self.map.contains_key(&id) { id = rand.gen();}`... I don't like the `loop` either.